scripts/zones/Riverne-Site_B01/mobs/Bahamut.lua

```diff
@@ -1,15 +1,97 @@
+    elseif (mobHPP < 50 and MegaFlareTrigger < 5) then
+        mob:setLocalVar("MegaFlareTrigger", 5);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 40 and MegaFlareTrigger < 6) then
+        mob:setLocalVar("MegaFlareTrigger", 6);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 30 and MegaFlareTrigger < 7) then
+        mob:setLocalVar("MegaFlareTrigger", 7);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 20 and MegaFlareTrigger < 8) then
+        mob:setLocalVar("MegaFlareTrigger", 8);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    end;
+
+    if (mob:actionQueueEmpty() == true and isBusy == false) then -- the last check prevents multiple Mega/Gigaflares from being called at the same time.
+        if (MegaFlareQueue > 0) then
+            mob:SetMobAbilityEnabled(false); -- disable all other actions until Megaflare is used successfully
+            mob:SetMagicCastingEnabled(false);
+            mob:SetAutoAttackEnabled(false);
+            if (FlareWait == 0 and tauntOneShown == 0 and tauntTwoShown == 0) then -- if there is a queued Megaflare and the last Megaflare has been used successfully or if the first one hasn't been used yet.
+                target:showText(mob,BAHAMUT_TAUNT);
+                mob:setLocalVar("FlareWait", mob:getBattleTime() + 2); -- second taunt happens two seconds after the first.
+                mob:setLocalVar("tauntOneShown", 1);
+            elseif (FlareWait < mob:getBattleTime() and FlareWait ~= 0 and tauntOneShown == 1) then -- the wait time between the first and second taunt as passed. Checks for wait to be not 0 because it's set to 0 on successful use.
+                if (tauntTwoShown == 0) then
+                    mob:setLocalVar("tauntTwoShown", 1); -- if Megaflare gets stunned it won't show the text again, until successful use.
+                    target:showText(mob,BAHAMUT_TAUNT + 1);
+                end;
+                if (mob:checkDistance(target) <= 15) then -- without this check if the target is out of range it will keep attemping and failing to use Megaflare. Both Megaflare and Gigaflare have range 15.
+                    if (mob:getBehaviour() == 1040) then -- default behaviour
+                        mob:setBehaviour(mob:getBehaviour() - 1024) -- 1024 is noturn
+                    end;
+                    mob:useMobAbility(1295);
+                end;    
+            end;    
+        elseif (MegaFlareQueue == 0 and mobHPP < 10 and GigaFlare < 1 and mob:checkDistance(target) <= 15) then  -- All of the scripted Megaflares are to happen before Gigaflare.
+            if (tauntThreeShown == 0) then
+                target:showText(mob,BAHAMUT_TAUNT + 2);
+                mob:setLocalVar("tauntThreeShown", 1); -- again, taunt won't show again until the move is successfully used.
+            end;
+            -- disable noturn here

```

Comment:
Oops I forgot to put in the noturn thing in here too. Copy the thing at line 80 and replace the comment.